### PR TITLE
Bug fix: Make id field unique

### PR DESCRIPTION
**Pull request by kyouscrap**
Name is the only unique field.

In a healthcare setting, there might be patients of the same name. To ensure that users can still add multiple patients with the same name, id field should be unique instead of name.

Let's make id field unique instead of name.

### CODE REVIEW

**Review by choonzies — COMMENTED** [review]
Implementation LGTM.

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "notes" : ""
   }, {
     "name": "Alice Pauline",
```
[the remark this answers is not in the document]
Comment (kyouscrap) [reply src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:10]: My idea behind the duplicatePersonAddressBook is more of handling the case where there is a duplicate entry of a person in the address book, i.e. all fields are the same.

**Review by clarud — COMMENTED** [review]
LGTM

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (choonzies) [reply src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:10]: Got it.

**Pull request merged**